FWCore/TestModules/plugins/StreamIDFilter.cc

```diff
@@ -0,0 +1,35 @@
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+namespace edmtest {
+  class StreamIDFilter : public edm::global::EDFilter<> {
+  public:
+    explicit StreamIDFilter(edm::ParameterSet const&);
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+    bool filter(edm::StreamID, edm::Event& e, edm::EventSetup const& c) const final;
+
+  private:
+    std::vector<unsigned int> rejectStreams_;
+  };
+
+  StreamIDFilter::StreamIDFilter(edm::ParameterSet const& ps)
+      : rejectStreams_(ps.getParameter<std::vector<unsigned int>>("rejectStreams")) {}
+
+  bool StreamIDFilter::filter(edm::StreamID id, edm::Event&, edm::EventSetup const&) const {
+    return std::find(rejectStreams_.begin(), rejectStreams_.end(), id) == rejectStreams_.end();
+  }
+
+  void StreamIDFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<std::vector<unsigned int>>("rejectStreams")
+        ->setComment("Stream IDs for which to reject events. If empty, all events are accepted.");
+    descriptions.addDefault(desc);
+  }
+}  // namespace edmtest
+
+DEFINE_FWK_MODULE(edmtest::StreamIDFilter);
```

Comment:
@Dr15Jones What convention do we want here? I can think of
* `edmtest::<module>`
* `edm::<module>`
* Have the `<module>` defined in some namespace, but define it to the framework via `<module>` type alias (i.e. without a namespace)

I started with the `edmtest::<module>`, but at this stage it would be easy to change.

Reply:
Or maybe other way to phrase the question would be if `edmtest` would be appropriate namespace for such generally-usable modules, or do we want to preserve it to modules that are specific for testing the framework?

Reply:
I think we want to use the `edmtest` for any useful module that is meant for testing and NOT meant for use in production. I also think `edmtest` should appear in the python configurations as it clearly shows this is a test.

Reply:
Thanks, so we can continue with my present approach.